src/main/java/com/netgrif/application/engine/workflow/service/DataService.java

```diff
@@ -308,6 +308,13 @@ private GetDataGroupsEventOutcome getDataGroups(String taskId, Locale locale, Se
         return groups;
     }
 
+    private void resolveTaskRefOrderOnGrid(DataGroup dataGroup, Map<String, Field> dataFieldMap) {
+        if (dataGroup.getLayout() != null && dataGroup.getLayout().getType() == "grid") {
+            List<Field> dataGroupFields = dataGroup.getData().stream().map(dataFieldMap::get).sorted(Comparator.comparingInt(a -> a.getLayout().getY())).collect(Collectors.toList());
+            dataGroup.setData(new LinkedHashSet<>(dataGroupFields.stream().map(Field::getStringId).collect(Collectors.toList())));
```

Comment:
why iterate over the stream twice? The `map` operator can be chained after the sorting and the desired set of IDs can then be collected with `Collectors.toCollection( LinkedHashSet::new )`